samples/azure/native-integration-asq/ASQN_10/Receiver/Program.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using NServiceBus;
+using NServiceBus.Azure.Transports.WindowsAzureStorageQueues;
+
+class Program
+{
+    static async Task Main()
+    {
+        var endpointName = "native-integration-asq";
+        Console.Title = endpointName;
+
+        var endpointConfiguration = new EndpointConfiguration(endpointName);
+        endpointConfiguration.UseSerialization<NewtonsoftSerializer>();
+        endpointConfiguration.EnableInstallers();
+
+        var transport = new AzureStorageQueueTransport("UseDevelopmentStorage=true");
+        // TODO: ASQ transport is not yet ready, Core v8 work is still in progress
+        // transport.DisablePublishing();
+        // transport.DelayedDelivery().DisableDelayedDelivery();
+
+        #region Native-message-mapping
+
+        transport.MessageUnwrapper = message => new MessageWrapper
+        {
+            Id = message.MessageId,
+            Body = message.Body.ToArray(),
+            Headers = new Dictionary<string, string>
+            {
+                {"NServiceBus.EnclosedMessageTypes", typeof(NativeMessage).FullName}
```

Comment:
Should we directly use the headers constant given that we reference NSB already?